### ISSUE TITLE
Additional symbolic expressions

### ISSUE THREAD

**Issue by dmtch** [illera88/Ponce#60]
Hi!
I was trying to solve [this crackme](http://ak42.io/2016/09/26/solving-a-crack-me-with-triton-and-pin-a-k-a-the-lazy-way/) with Ponce. In this way I got a problem. Look:
1. run the binary with one argument "AAAAAAAAAAAA", stop on main() and symbolize argv[1] bytes;
2. then, stop on first branch that checks argv[1] characters and use "Negate & Inject" feature;
3. look at argv[1] - it's filled with invalid characters, e.g. NULLs.
For this challenge we need to put additional expressions for argv[1] variables: they have to be in range from 0x20 to 0x7f, obviously :)
Is there any way to do this?
Thanks!

**Comment by 0ca** [comment]
Hi!

Thanks for trying Ponce. I check the same crackme with the same result:
```
[+] Solution found! Values:
 - SymVar_1 (argv[1][1]):0xff 
 - SymVar_3 (argv[1][3]):00 
 - SymVar_7 (argv[1][7]):00
 - SymVar_9 (argv[1][9]):00
```

We could add an option to automatically add an ascii constrain to all the characters in the arguments.

But it should be an option so the user can solve conditions that need the argument to be shorter, like in:

````
if strlen(argv[1])>5)
{
     printf("Too long!\n");
     exit(-1);
}
```

I will try to implement this soon.

Thank you for the feedback ;)

**Comment by dmtch** [comment]
Maybe it will be more universal if we could set arbitrary expressions for our variables, not just ascii characters for argv? What do you think?

**Comment by 0ca** [comment]
I agree. It would be cool to be able to set "arbitrary expressions", or constraints to variable in the "Taint/Symbolic items" window:
![image](https://cloud.githubusercontent.com/assets/5008707/19359979/5ae46970-9131-11e6-9c61-b39358714519.png)

**Comment by jhonMalcom79** [comment]
implemented at 23ea73e

**Comment by illera88** [comment]
Implemented at 23ea73e
